carp.common/src/commonMain/kotlin/dk/cachet/carp/common/ExtractUniqueKeyMap.kt

```diff
@@ -2,7 +2,8 @@ package dk.cachet.carp.common
 
 
 /**
- * A [Map] which automatically extracts the key of added values and throws an error when attempting to add an element with a key which is already present.
+ * A [Map] which automatically extracts the key of added values and
```

Comment:
I don't fully understand the need for this construct. The semantics of putting an existing key into a mutable map is overwriting. If you don't want to able to overwrite keys you can use an immutable map. If what you want is really an "append only mutable map" then that should be its name: `AppendOnlyMutableMap`.

An error free version could look something like this:
`class AppendOnlyMap<K, V> : LinkedHashMap<K, V>() {
    override fun put(key: K, value: V): V? {
       return putIfAbsent(key, value)
    }

    override fun putAll(from: Map<out K, V>) {
        from.forEach { (k, u) -> putIfAbsent(k, u) }
    }
}`